### DISCUSSION TITLE
Leantime V2.1.9

### DISCUSSION THREAD

**Discussion by marcelfolaron** [Leantime/leantime#685] [Announcements]
<hr /><em>This discussion was created from the release <a href='https://github.com/Leantime/leantime/releases/tag/'>Leantime V2.1.9</a>.</em>

**Comment by marcelfolaron** [comment]
What a release! Thanks everyone who contributed!

* Leantime is now available in 15 languages!!!
* We added simple ldap support. Head to the config file and try it out!
* So. Many. Bug. Fixes. Thank you!

**Reply by mshaw-80** [reply]
Thanks for updates. Appreciate all the effort that went into it.

I've installed the latest version and have come across an issue that I wasn't having before. Thats is, whenever I save an item (task, milestone etc.), I'm met with a http error 500 page.

Any idea what's causing this?

**Reply by joaobrazuna** [reply]
> * Leantime is now available in 15 languages!!!

Hello,
I've translate all the messages (v2.19) to European Portuguese (Portugal), and using it now.
Anyone can commit the files to me?

**Comment by marcelfolaron** [comment]
Oh no, what errors do you see in resources/error.log ?

**Reply by mshaw-80** [reply]
Hopefully this makes sense:

Edited. Error message removed.

**Comment by marcelfolaron** [comment]
Can you try to set the config variable:

$email to a correct email return address?
That should fix it. I still need to fail more gracefully in that case.

**Reply by mshaw-80** [reply]
Thats fixed it. Thank you so much.
Leantime is an awesome system by the way. I love it.

**Reply by brozena** [reply]
Worked for me too, fyi. :)

**Comment by marcelfolaron** [comment]
Thank you! Appreciate it. Will work on a graceful fallback for this issue

**Comment by unknown797** [comment]
Hi Team,

Love the new fixes on 2.1.9 (especially the password issue), but I'm also having the same issues with saving or submitting any changes on any pages it tries to redirect to the same page, but with an error 500 on Chrome and no error messages on Firefox just blank.

I've re-tested this on 2.1.8 and it appears to be working correctly using the subfolder installation method. I also don't have email setup as its for my own personal use, so I don't require an email notification. What value should I set $email to in my case?

Uncaught PHPMailer\PHPMailer\Exception: Invalid address:  (From):  in \vendor\phpmailer\phpmailer\src\PHPMailer.php:1300

Thanks

**Comment by marcelfolaron** [comment]
Hi,
I updated phpmailer and apparently it is not very happy with empty from
addresses anymore. Can you make sure the email variable in the config file
is set? That should fix it. I am working on a fix for that.

**Comment by marcelfolaron** [comment]
Sorry, you can set it to any address. Even just ***@***.***

It’s not validating whether you are allowed to send from that address, just
that it exists

**Reply by unknown797** [reply]
Thank you so much @marcelfolaron.

For those that don't know code:

/config/configuration.php

line 39

public $email = "xxx@xxx.xxx"; //Return email address

**Comment by paschauf** [comment]
Hi guys, I am frustrated after I updated to Leantime 2.1.9, with PHP 8.1, MariaDB 10.6.7. I run Leantime on a cloud server in an LXC container with Ubuntu 22.04 LTS and an nginx reverse proxy (in another lxc container). Everything worked fine before but after the upgrade, when I login I keep staying on the login page. There are some warnings on not being able to change sessions and a proxy_fcgi error. Any good soul can point me on the right direction on what to change? If I type a wrong username/password I get the correct error message, if I input the correct user name/password it just redirects to the login page without any warning and I am stuck there.  

Leantime Installed on /var/www/leantime
Apache2 points to /var/www/leantime/public/
```
<VirtualHost *:80>
    ServerAdmin admin@***.com
    DocumentRoot /var/www/leantime/public/
    ServerName  leantime.***.com

    <Directory /var/www/leantime/public/>
        Options FollowSymLinks
        AllowOverride All
        Order allow,deny
        allow from all
    </Directory>

    <FilesMatch \.php$>
        # From the Apache version 2.4.10 and above, use the SetHandler to run PHP as a fastCGI process server
         SetHandler "proxy:unix:/run/php/php8.1-fpm.sock|fcgi://localhost"
    </FilesMatch>

    ErrorLog /var/log/leantime/leantime-error.log
    CustomLog /var/log/leantime/leantime-access.log common
</VirtualHost>
```



I use php8.1-fpm and Fastcgi with a weird error that the pdo_mysql cannot be found but it's actually there 
```
● php8.1-fpm.service - The PHP 8.1 FastCGI Process Manager
     Loaded: loaded (/lib/systemd/system/php8.1-fpm.service; enabled; vendor preset: enabled)
    Drop-In: /run/systemd/system/service.d
             └─zzz-lxc-service.conf
     Active: active (running) since Mon 2022-06-20 14:32:28 HKT; 1min 38s ago
       Docs: man:php-fpm8.1(8)
    Process: 1242 ExecStartPost=/usr/lib/php/php-fpm-socket-helper install /run/php/php-fpm.sock /etc/php/8.1/fpm/pool.d/www.conf 81 (code=exited, status=0/SUCCESS)
   Main PID: 1239 (php-fpm8.1)
     Status: "Processes active: 0, idle: 2, Requests: 5, slow: 0, Traffic: 0req/sec"
      Tasks: 3 (limit: 4580)
     Memory: 15.0M
        CPU: 254ms
     CGroup: /system.slice/php8.1-fpm.service
             ├─1239 "php-fpm: master process (/etc/php/8.1/fpm/php-fpm.conf)" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" ""
             ├─1240 "php-fpm: pool www" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" ""
             └─1241 "php-fpm: pool www" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" "" ""
Jun 20 14:32:28 leantime systemd[1]: Starting The PHP 8.1 FastCGI Process Manager...
Jun 20 14:32:28 leantime php-fpm8.1[1239]: [20-Jun-2022 14:32:28] NOTICE: PHP message: PHP Warning:  PHP Startup: Unable to load dynamic library 'pdo_mysql' (tried: /usr/lib/php/20210902/pdo_mysql (/usr/lib/php/20210902/pdo_mysql: cannot open shared object file: No such file o>
Jun 20 14:32:28 leantime php-fpm8.1[1239]: [20-Jun-2022 14:32:28] NOTICE: PHP message: PHP Warning:  Module "imap" is already loaded in Unknown on line 0
Jun 20 14:32:28 leantime systemd[1]: Started The PHP 8.1 FastCGI Process Manager.
```



Log 1 (/var/www/leantime/resources/logs/error.log): 
```
[20-Jun-2022 14:00:39 Asia/Hong_Kong] PHP Warning:  ini_set(): Session ini settings cannot be changed when a session is active in /var/www/leantime-2.1.9/src/core/class.session.php on line 41
[20-Jun-2022 14:00:39 Asia/Hong_Kong] PHP Deprecated:  explode(): Passing null to parameter #2 ($string) of type string is deprecated in /var/www/leantime-2.1.9/src/core/class.session.php on line 52
[20-Jun-2022 14:00:39 Asia/Hong_Kong] PHP Warning:  session_name(): Session name cannot be changed when a session is active in /var/www/leantime-2.1.9/src/core/class.session.php on line 71
[20-Jun-2022 14:00:39 Asia/Hong_Kong] PHP Warning:  session_id(): Session ID cannot be changed when a session is active in /var/www/leantime-2.1.9/src/core/class.session.php on line 72
[20-Jun-2022 14:00:39 Asia/Hong_Kong] PHP Notice:  session_start(): Ignoring session_start() because a session is already active in /var/www/leantime-2.1.9/src/core/class.session.php on line 73
```

Log 2 (leantime-error.log):
```
[Mon Jun 20 14:00:39.250070 2022] [proxy_fcgi:error] [pid 900] [client 180.162.72.214:0] AH01071: Got error '
PHP message: PHP Warning:  ini_set(): Session ini settings cannot be changed when a session is active in /var/www/leantime-2.1.9/config/settings.php on line 49
PHP message: PHP Warning:  ini_set(): Session ini settings cannot be changed when a session is active in /var/www/leantime-2.1.9/config/settings.php on line 50
PHP message: PHP Warning:  ini_set(): Session ini settings cannot be changed when a session is active in /var/www/leantime-2.1.9/config/settings.php on line 51
PHP message: PHP Warning:  ini_set(): Session ini settings cannot be changed when a session is active in /var/www/leantime-2.1.9/config/settings.php on line 52', referer: http://leantime.********.com/
```

```
[PHP Modules]
calendar
Core
ctype
curl
date
exif
FFI
fileinfo
filter
ftp
gd
gettext
hash
iconv
imap
json
libxml
mbstring
mysqli
mysqlnd
openssl
pcntl
pcre
PDO
pdo_mysql
Phar
posix
readline
Reflection
session
shmop
sockets
sodium
SPL
standard
sysvmsg
sysvsem
sysvshm
tokenizer
xmlrpc
Zend OPcache
zlib

[Zend Modules]
Zend OPcache
```

Apache modules enabled:
`access_compat actions alias auth_basic authn_core authn_file authz_core authz_host authz_user autoindex deflate dir env fcgid filter mime mpm_prefork negotiation php8.1 proxy proxy_fcgi proxy_fdpass remoteip reqtimeout rewrite session session_cookie setenvif status`


Extract of leantime/config/configuration.php:
...
```
   /* General */
    public $sitename = "Leantime"; //Name of your site, can be changed later
    public $language = "en-US"; //Default language
    public $mainColor = "1b75bb"; //Default color, can be changed later
    public $logoPath = "/images/logo.png"; //Default logo path, can be changed later
    public $appUrl = ""; //Base URL, trailing slash not needed
    public $defaultTimezone = "Asia/Hong_Kong"; //Set default timezone

    /* Database */
    public $dbHost="localhost"; //Database host
    public $dbUser="***"; //Database username
    public $dbPassword="***"; //Database password
    public $dbDatabase="***"; //Database name

    /* Fileupload */
    public $userFilePath= "userfiles/"; //Local relative path to store uploaded files (if not using S3)

    public $dbBackupPath= "backupdb/"; //Local relative path to store backup files, need permission to write

    public $useS3 = false; //Set to true if you want to use S3 instead of local files
    public $s3Key = ""; //S3 Key
    public $s3Secret = ""; //S3 Secret
    public $s3Bucket = ""; //Your S3 bucket
    public $s3UsePathStyleEndpoint = false; // false => https://[bucket].[endpoint] ; true => https://[endpoint]/[bucket]
    public $s3Region = ""; //S3 region
    public $s3FolderName = ""; //Foldername within S3 (can be emtpy)
    public $s3EndPoint = ""; //S3 EndPoint S3 Compatible (https://sfo2.digitaloceanspaces.com)

    /* Sessions */
    public $sessionpassword = "***"; //Salting sessions. Replace with a strong password
    public $sessionExpiration = 28800; //How many seconds after inactivity should we logout?  28800seconds = 8hours
```
...



Any hint really appreciated, thanks in advance!

**Reply by gloriafolaron** [reply]
@paschauf have you had any luck on this?

**Reply by paschauf** [reply]
Hi Gloria, nope ... still redirecting to the login page ... considering reverting back to an older Ubuntu and PHP if I can't identify the problem...

**Reply by paschauf** [reply]
OK lol, after getting a massive headache over several days trying to reinstall everything (tried without php-fpm and with more simple configs, checking rewrite mods and .htaccess) from scratch it came down to a browser problem. For some reason everything works with Edge but not with my favorite browser Vivaldi (based on chromium). The user doesn't get passed on with Vivaldi and it stays on the login page. Anyhow, with Edge everything works with Ubuntu 22.04 Jammy inside an lxc container, Apache2, PHP 8.1, and MariaDB behind a NGINX reverse proxy. 

Anyone has an idea on what could be the issue with Vivaldi?

**Reply by gloriafolaron** [reply]
@paschauf I’m not as familiar with Vivaldi as I should for the use case.  @marcelfolaron , I don’t believe this is one we currently do any testing with.  Any thoughts?

**Reply by paschauf** [reply]
I suspect cookies/php session related

**Reply by onyxnz** [reply]
Vivaldi is still at issue here for me, too. It is non-functional, in that it keeps looping the login system. I now have Chromium, FF, Midori, Falkon are working on the same computer without issue. I was on Discord today with Marcel, looking at this.
It's not cookies, I cleared those, it's not Vivaldi's ad blockers, I turned everything off...thankfully I have plenty of browsers to choose from, but Vivaldi has been my main one for years now, and this is the first issue it's had like this.
Leantime 2.3.23. Ubu 22.04/PHP8.1-FPM/Nginx/MariaDB on server side. Archlinux 64bit on client side.

**Reply by onyxnz** [reply]
It's only Vivaldi on Linux. I've opened a thread in their forum...
[https://forum.vivaldi.net/topic/87128/leantime-problems](https://forum.vivaldi.net/topic/87128/leantime-problems)

**Comment by marcelfolaron** [comment]
Hey,

I think the first warning is probably the main culprit. Something is up
with you pdo_mysql module.

Php says it can’t load it. I believe everything else following that error
message is a direct result of that issue (session started when error
message appears, so a new Leantime session cannot  be started, ini_set
won’t work when a session is in progress etc)

Can you check if the pdo_mysql module is actually installed? It shows up in
your list of modules but the files might not be there or are corrupt.
